crates/net/downloaders/src/bodies/bodies.rs

```diff
@@ -225,13 +226,16 @@ where
         self.metrics.buffered_responses.decrement(1.);
         self.buffered_blocks_size_bytes -= resp.size();
         self.metrics.buffered_blocks.decrement(resp.len() as f64);
-        self.metrics.buffered_blocks_size_bytes.set(resp.size() as f64);
+        self.metrics.buffered_blocks_size_bytes.set(self.buffered_blocks_size_bytes as f64);
         Some(resp)
     }
 
     /// Adds a new response to the internal buffer
     fn buffer_bodies_response(&mut self, response: Vec<BlockResponse>) {
-        let size = response.iter().map(|b| b.size()).sum::<usize>();
+        // take into account capacity
+        let size = response.iter().map(BlockResponse::size).sum::<usize>() +
+            (response.capacity() - response.len()) * mem::size_of::<BlockResponse>();
```

Comment:
do we want a helper for `v: Vec<T>` which does `(v.capacity() - v.len()) * size_of<T>`